File: VB/HyperLinkPropertyEditor.Win/GridListEditorOpenHyperLinkViewController.vb
Imports System
Imports System.Drawing
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo

Namespace HyperLinkPropertyEditor.Win
    Public Class GridListEditorOpenHyperLinkViewController
        Inherits ViewController(Of ListView)

        Private gridListEditor As GridListEditor
        Protected Overrides Sub OnViewControlsCreated()
            MyBase.OnViewControlsCreated()
            gridListEditor = TryCast(CType(View, ListView).Editor, GridListEditor)
            If gridListEditor IsNot Nothing AndAlso gridListEditor.GridView IsNot Nothing Then
                AddHandler gridListEditor.GridView.MouseDown, AddressOf GridView_MouseDown
            End If
        End Sub
        Protected Overrides Sub OnDeactivated()
            If gridListEditor IsNot Nothing AndAlso gridListEditor.GridView IsNot Nothing Then
                RemoveHandler gridListEditor.GridView.MouseDown, AddressOf GridView_MouseDown
            End If
            MyBase.OnDeactivated()
        End Sub
        Private Sub GridView_MouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs)
            Dim gv As GridView = DirectCast(sender, GridView)
            Dim hi As GridHitInfo = gv.CalcHitInfo(New Point(e.X, e.Y))
            If hi.InRowCell Then
                Dim repositoryItemHyperLinkEdit As RepositoryItemHyperLinkEdit = TryCast(hi.Column.ColumnEdit, RepositoryItemHyperLinkEdit)
                If repositoryItemHyperLinkEdit IsNot Nothing Then
                    Dim editor As HyperLinkEdit = CType(repositoryItemHyperLinkEdit.CreateEditor(), HyperLinkEdit)
                    editor.ShowBrowser(WinHyperLinkStringPropertyEditor.GetResolvedUrl(gv.GetRowCellValue(hi.RowHandle, hi.Column)))
                End If
            End If
        End Sub
    End Class
End Namespace

File: CS/HyperLinkPropertyEditor.Win/GridListEditorOpenHyperLinkViewController.cs
using System;
using System.Drawing;
using DevExpress.ExpressApp;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraEditors.Repository;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;

namespace HyperLinkPropertyEditor.Win {
	public class GridListEditorOpenHyperLinkViewController : ViewController<ListView> {
		private GridListEditor gridListEditor;
		protected override void OnViewControlsCreated() {
			base.OnViewControlsCreated();
			gridListEditor = ((ListView)View).Editor as GridListEditor;
			if(gridListEditor != null && gridListEditor.GridView != null) {
				gridListEditor.GridView.MouseDown += GridView_MouseDown;
			}
		}
		protected override void OnDeactivated() {
			if(gridListEditor != null && gridListEditor.GridView != null) {
				gridListEditor.GridView.MouseDown -= GridView_MouseDown;
			}
			base.OnDeactivated();
		}
		private void GridView_MouseDown(object sender, System.Windows.Forms.MouseEventArgs e) {
			GridView gv = (GridView)sender;
			GridHitInfo hi = gv.CalcHitInfo(new Point(e.X, e.Y));
			if(hi.InRowCell) {
				RepositoryItemHyperLinkEdit repositoryItemHyperLinkEdit = hi.Column.ColumnEdit as RepositoryItemHyperLinkEdit;
				if(repositoryItemHyperLinkEdit != null) {
					HyperLinkEdit editor = (HyperLinkEdit)repositoryItemHyperLinkEdit.CreateEditor();
					editor.ShowBrowser(WinHyperLinkStringPropertyEditor.GetResolvedUrl(gv.GetRowCellValue(hi.RowHandle, hi.Column)));
				}
			}
		}
	}
}